base/linalg/diagonal.jl

```diff
@@ -266,6 +266,30 @@ function A_ldiv_B!(D::Diagonal{T}, V::AbstractMatrix{T}) where T
     V
 end
 
+Ac_ldiv_B!(D::Diagonal{T}, B::AbstractVecOrMat{T}) where T = A_ldiv_B!(conj(D), B)
+At_ldiv_B!(D::Diagonal{T}, B::AbstractVecOrMat{T}) where T = A_ldiv_B!(D, B)
+
+function A_rdiv_B!(A::AbstractMatrix{T}, D::Diagonal{T}) where T
+    dd = D.diag
+    m, n = size(A)
+    if (k = length(dd)) ≠ n
+        throw(DimensionMismatch("left hand side has $n columns but D is $k by $k"))
+    end
+    @inbounds for j in 1:n
+        ddj = dd[j]
+        if iszero(ddj)
+            throw(SingularException(j))
+        end
+        for i in 1:m
+            A[i, j] /= ddj
```

Comment:
See @dmbates's top post. Using division is less prone to overflow e.g.
```julia
julia> 1e-100/1e-310
1.0000000000000031e210
```
vs
```julia
julia> 1e-100*inv(1e-310)
Inf
```
and is only one rounding instead of two. Running with `--math-mode=fast` allows for this rewrite so maybe we should use division.